android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -752,6 +759,8 @@ private void checkSendCrashReport() {
     KMManager.setMaySendCrashReport(maySendCrashReport);
   }
 
+  private Uri requestPermissionIntentUri;
```

Comment:
This class variable replaces `data` -- and is used in precisely two places below, which is a whole lot less messy.